proposals/explicit-backing-fields.md

```diff
@@ -0,0 +1,317 @@
+# Explicit Backing Fields
+
+- **Type**: Design Proposal
+- **Author**: ?
+- **Contributors**: Svetlana Isakova, Kirill Rakhman, Dmitry Petrov, Roman Elizarov, Ben Leggiero, Matej Drobnič, Mikhail Glukhikh, Nikolay Lunyak
+- **Status**: Implemented in FIR
+- **Prototype**: Implemented
+- **Initial YouTrack Issue**: [KT-14663](https://youtrack.jetbrains.com/issue/KT-14663)
+- **Initial Proposal**: [private_public_property_types#122](https://github.com/Kotlin/KEEP/pull/122)
+
+## Summary
+
+**Note**: initial proposal contents have been partially copied down here for convenience. Despite the different approach, the already shown use cases are still relevant.
+
+>  Common pattern in java is having full type accessible as private property and then only exposing required interface in the public getter:
+>
+> ```java
```

Comment:
Personally I think that Java example is good, but may be it shouldn't be in the title section (put it along with "Android architecture components"). About this place, I'd try to explain this pattern in pure Kotlin without direct mentioning of Java. @elizarov , do you have some other ideas here?

Reply:
+1. We should cater to Kotlin developer here. Frankly, I don't think that comparison with Java is actually useful in this document at all. Kotlin already has all the concepts: properties and backing field. Moreover, there is already an existing pattern of a "backing properties" that this design is striving to improve https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties